node/core/dispute-coordinator/src/import.rs

```diff
@@ -179,35 +165,43 @@ impl CandidateVoteState<CandidateVotes> {
 	pub fn new_from_receipt(candidate_receipt: CandidateReceipt) -> Self {
 		let votes =
 			CandidateVotes { candidate_receipt, valid: BTreeMap::new(), invalid: BTreeMap::new() };
-		Self {
-			votes,
-			own_vote: OwnVoteState::NoVote,
-			concluded_invalid: false,
-			concluded_valid: false,
-			is_confirmed: false,
-			is_disputed: false,
-		}
+		Self { votes, own_vote: OwnVoteState::NoVote, dispute_status: None }
 	}
 
 	/// Create a new `CandidateVoteState` from already existing votes.
-	pub fn new<'a>(votes: CandidateVotes, env: &CandidateEnvironment<'a>) -> Self {
+	pub fn new<'a>(votes: CandidateVotes, env: &CandidateEnvironment<'a>, now: Timestamp) -> Self {
 		let own_vote = OwnVoteState::new(&votes, env);
 
 		let n_validators = env.validators().len();
 
 		let supermajority_threshold =
 			polkadot_primitives::v2::supermajority_threshold(n_validators);
 
-		let concluded_invalid = votes.invalid.len() >= supermajority_threshold;
-		let concluded_valid = votes.valid.len() >= supermajority_threshold;
-
 		// We have a dispute, if we have votes on both sides:
 		let is_disputed = !votes.invalid.is_empty() && !votes.valid.is_empty();
 
-		let byzantine_threshold = polkadot_primitives::v2::byzantine_threshold(n_validators);
-		let is_confirmed = votes.voted_indices().len() > byzantine_threshold && is_disputed;
+		let dispute_status = if is_disputed {
+			let mut status = DisputeStatus::active();
+			let byzantine_threshold = polkadot_primitives::v2::byzantine_threshold(n_validators);
+			let is_confirmed = votes.voted_indices().len() > byzantine_threshold && is_disputed;
```

Comment:
Isn't `is_disputed` redundant here?

Reply:
Something is only disputed if there are votes on both sides. That check was missing previously in one place, which caused the dispute coordinator to record one sided disputes.

Reply:
This is happening in vote import, which is called on every vote import, not just the one coming from dispute-distribution. Indeed this is what makes this code so tricky, everybody is in the mindset of disputes, but the mere fact that we are importing votes does not necessarily mean there is a dispute - it can also just be the import of backing votes, in case there will be a later dispute. With small test networks, where the network only consists of the backing group - there is no approval voting and the number of backing votes suffices to reach super majority. Hence if we don't check that there are opposing votes as well, we would assume a dispute has concluded, while in reality there was no dispute to begin with.

Reply:
> Something is only disputed if there are votes on both sides. That check was missing previously in one place, which caused the dispute coordinator to record one sided disputes.

I meant the `&& is_disputed` on line 186. If we are there it should always be true?

Reply:
Haha, yeah that one is redundant indeed.